### PR TITLE
feat: initial script for locking half

### PR DESCRIPTION
**Pull request by GalloDaSballo**
Lock half of available (in strategy) CVX
@gosuto-inzasheru @petrovska-petro @Tritium-VLK 

Note: Because the code I wrote for the strategy is pretty obtuse, we have up to 1% rounding

### CODE REVIEW

**Comment by GalloDaSballo** [comment]
Test with
```
brownie run scripts/issue/643/lock_half_of_cvx.py
```

File: scripts/issue/643/lock_half_of_cvx.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    ## We want to lock half of what's in the Strategy
+    
+
+
+    ## Our Target is 50% of the CVX in the strat
+    target_amount = balanceOfWant // 2
+
+    ## in BPS
+    as_bps = (target_amount + balanceInLock) * 10_000 // totalCVXBalance
```
Comment (sajanrajdev) [inline scripts/issue/643/lock_half_of_cvx.py:53]: Why not using `MAX_BPS` here?

File: scripts/issue/643/lock_half_of_cvx.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    
+    ## Lock
+    safe.badger.strat_bvecvx.manualRebalance(as_bps)
+
+    after_balance_in_vault = safe.convex.cvx.balanceOf(registry.eth.sett_vaults.bveCVX)
+    after_balance_of_locker = safe.convex.cvx.balanceOf(safe.badger.strat_bvecvx.LOCKER())
+
+    assert approx(after_balance_in_vault - before_balance_in_vault, expected_in_vault, 1)
+    assert approx(after_balance_of_locker - before_balance_of_locker, cvxToLock, 1)
```
Comment (sajanrajdev) [inline scripts/issue/643/lock_half_of_cvx.py:90]: You will need to add `safe.post_safe_tx()` at the end of the script in order to post the tx.

**Comment by sajanrajdev** [comment]
Estimates the percentage on CVX (in bps) to be locked and calls manual rebalance with it. Math checks out with what the function is expecting and will lead to half of the recently unlocked CVX to be re-locked.

Only left a couple of comments, otherwise it LG

File: scripts/issue/643/lock_half_of_cvx.py (hunk 2 above)
Comment (gosuto-inzasheru) [inline scripts/issue/643/lock_half_of_cvx.py:90]:
```suggestion
    assert approx(after_balance_of_locker - before_balance_of_locker, cvxToLock, 1)
    
    safe.post_safe_tx()
```

File: scripts/issue/643/lock_half_of_cvx.py (hunk 1 above)
Comment (gosuto-inzasheru) [inline scripts/issue/643/lock_half_of_cvx.py:53]:
```suggestion
    as_bps = (target_amount + balanceInLock) * MAX_BPS // totalCVXBalance 
```

**Review by gosuto-inzasheru — COMMENTED** [review]
looks good, result in trace:
```
    ├── CvxLockerV2.lock  [CALL]  1469:6751  [163681 / 192874 gas]
    │   │   ├── address: 0x72a19342e8F1838460eBFCCEf09F6585e32db86E
    │   │   ├── value: 0
    │   │   └── input arguments:
    │   │       ├── _account: 0x898111d1F4eB55025D0036568212425EE2274082
    │   │       ├── _amount: 828149162369194015064963
    │   │       └── _spendRatio: 0
    │   │   
    │   │   
    │   ├── ERC20.transferFrom  [CALL]  4152:4637  [23491 gas]
    │   │       ├── address: 0x4e3FBD56CD56c3e72c1403e103b45Db9da5B9D2B
    │   │       ├── value: 0
    │   │       └── input arguments:
    │   │           ├── _from: 0x898111d1F4eB55025D0036568212425EE2274082
    │   │           ├── _to: 0x72a19342e8F1838460eBFCCEf09F6585e32db86E
    │   │           └── _value: 828149162369194015064963
```

we will just have to make sure we turn off the `earn` call from the bots for that week, as to not also automatically lock the remainder..

**Comment by dapp-whisperer** [comment]
like the balance checks after operation. need to review those function calls.
@btcookies can we ensure earn does not run this week?

**Comment by GalloDaSballo** [comment]
Generally speaking I think it's best if you commit non-critical changes as the overhead of making the comment, reviewing and then re-reviewing is way higher than just pushing a fix

**Review by gosuto-inzasheru — APPROVED** [review]

**Comment by GalloDaSballo** [comment]
Please @sajanrajdev @gosuto-inzasheru @dapp-whisperer Copy Paste the Script into a new Branch and PR it, will set up keys for next time

**Comment by GalloDaSballo** [comment]
@gosuto-inzasheru Check now

**Comment by gosuto-inzasheru** [comment]
yes you got it! the only thing is, in order to merge all commits in the branch need to be signed 🥲

**Pull request closed without merging**